src/model/GroundHeatExchangerHorizontalTrench.cpp

```diff
@@ -510,27 +515,45 @@ namespace model {
   }
 
   bool GroundHeatExchangerHorizontalTrench::setGroundTemperatureModel(const std::string& groundTemperatureModel) {
-    return getImpl<detail::GroundHeatExchangerHorizontalTrench_Impl>()->setGroundTemperatureModel(groundTemperatureModel);
+    LOG(Warn, "As of 3.6.0, setGroundTemperatureModel is deprecated. Use undisturbedGroundTemperatureModel instead. It will be removed within three "
+              "releases.");
+    return false;
```

Comment:
maybe we do a check on undisturbedGroundTemperatureModel().iddObjectType() and see if it matches perhaps?